crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -165,8 +173,11 @@ pub(crate) struct TypeInference<'db> {
     /// The types of every expression in this region.
     expressions: FxHashMap<ScopedExpressionId, Type<'db>>,
 
-    /// The types of every definition in this region.
-    definitions: FxHashMap<Definition<'db>, Type<'db>>,
+    /// The types of every binding in this region.
+    bindings: FxHashMap<Definition<'db>, Type<'db>>,
```

Comment:
I'm surprised that adding an extra hash map doesn't lead to a more significant perf regeression. Lucky us.